libs/bare/basic.ts

```diff
@@ -1,16 +1,16 @@
 namespace control {
     //% shim=serial::writeString
-    export function __log(s: string) {
+    export function __log(s: any) {
     }
 
-    export function dmesg(s: string) {
+    export function dmesg(s: any) {
```

Comment:
same as __log, C++ expects a string.

Reply:
The thumb code emitter will emit conversion to string regardless of TS type. There is however dmesgVal or something like that which will print more info about non strings.